lib/temporal/workflow/execution_info.rb

```diff
@@ -1,37 +1,20 @@
 require 'temporal/concerns/payloads'
+require 'temporal/workflow/status'
 
 module Temporal
   class Workflow
     class ExecutionInfo < Struct.new(:workflow, :workflow_id, :run_id, :start_time, :close_time, :status, :history_length, :memo, keyword_init: true)
       extend Concerns::Payloads
 
-      RUNNING_STATUS = :RUNNING
-      COMPLETED_STATUS = :COMPLETED
-      FAILED_STATUS = :FAILED
-      CANCELED_STATUS = :CANCELED
-      TERMINATED_STATUS = :TERMINATED
-      CONTINUED_AS_NEW_STATUS = :CONTINUED_AS_NEW
-      TIMED_OUT_STATUS = :TIMED_OUT
-
-      API_STATUS_MAP = {
-        WORKFLOW_EXECUTION_STATUS_RUNNING: RUNNING_STATUS,
-        WORKFLOW_EXECUTION_STATUS_COMPLETED: COMPLETED_STATUS,
-        WORKFLOW_EXECUTION_STATUS_FAILED: FAILED_STATUS,
-        WORKFLOW_EXECUTION_STATUS_CANCELED: CANCELED_STATUS,
-        WORKFLOW_EXECUTION_STATUS_TERMINATED: TERMINATED_STATUS,
-        WORKFLOW_EXECUTION_STATUS_CONTINUED_AS_NEW: CONTINUED_AS_NEW_STATUS,
-        WORKFLOW_EXECUTION_STATUS_TIMED_OUT: TIMED_OUT_STATUS
-      }.freeze
-
-      VALID_STATUSES = [
-        RUNNING_STATUS,
-        COMPLETED_STATUS,
-        FAILED_STATUS,
-        CANCELED_STATUS,
-        TERMINATED_STATUS,
-        CONTINUED_AS_NEW_STATUS,
-        TIMED_OUT_STATUS
-      ].freeze
+      STATUSES = [
+        Temporal::Workflow::Status::RUNNING,
```

Comment:
@antstorm I haven't synced the `running` => `open` rename or `closed` status addition yet. I'm not sure how to represent the `closed` status as a `Temporal::Api::Enums::V1::WorkflowExecutionStatus` value. Any thoughts here?

[Cadence change](https://github.com/coinbase/cadence-ruby/pull/66/files#diff-8930818514d8289103861231a9094ed0b30609e3365da38f0f3da35b827b7282R8) for reference

Reply:
`closed` is not represented in Thrift or Protos, it's a convenience status, which just translates to `!running`.

As for `running` you can leave it as is in Temporal since it references a Proto status (Cadence doesn't have explicit status for running workflows)

Reply:
Sounds good, thanks!